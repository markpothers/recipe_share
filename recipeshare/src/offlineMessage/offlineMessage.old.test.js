import React from "react";
import renderer, { act } from "react-test-renderer";

import OfflineMessage from "./offlineMessage"
import { TouchableOpacity } from "react-native";

describe("OfflineMessage", () => {

	let component

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(() => {
		// console.log('runs before every test')
		jest.useFakeTimers()
	})

	afterEach(() => {
		// console.log('runs after each test')
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	test("renders with an action button", () => {
		let mockClearOfflineMessage = jest.fn()
		let mockAction = jest.fn()
		act(() => {
			component = renderer.create(
				<OfflineMessage
					clearOfflineMessage={mockClearOfflineMessage}
					message={"Sorry, you can't do that right now, you appear to be offline"}
					action={mockAction}
				/>
			)
		})
		const image = component.toJSON()
		expect(image).toMatchSnapshot()
		let root = component.root
		const buttons = root.findAllByType(TouchableOpacity)
		expect(buttons.length).toEqual(1)
	})

	test("renders without an action button", () => {
		let mockClearOfflineMessage = jest.fn()
		act(() => {
			component = renderer.create(
				<OfflineMessage
					clearOfflineMessage={mockClearOfflineMessage}
					message={"Sorry, you can't do that right now, you appear to be offline"}
				/>
			)
		})
		const image = component.toJSON()
		expect(image).toMatchSnapshot()
		let root = component.root
		const buttons = root.findAllByType(TouchableOpacity)
		expect(buttons.length).toEqual(0)
	})

});
