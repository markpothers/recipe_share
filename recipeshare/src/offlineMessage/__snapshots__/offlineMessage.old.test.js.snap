// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfflineMessage renders with an action button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#104e01",
      "borderColor": "#fff59b",
      "borderRadius": 11.25,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 0,
      "position": "absolute",
      "top": undefined,
      "zIndex": 2,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "opacity": 1,
      }
    }
  >
    <Text
      maxFontSizeMultiplier={2.5}
      style={
        Object {
          "color": "#fff59b",
          "fontSize": 30.595932917809698,
          "paddingHorizontal": 15,
          "paddingVertical": 13.34,
          "textAlign": "center",
        }
      }
    >
      Sorry, you can't do that right now, you appear to be offline
    </Text>
    <Icon
      name="forwardburger"
      size={66.7}
      style={
        Object {
          "color": "#fff59b",
          "marginRight": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`OfflineMessage renders without an action button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#104e01",
      "borderColor": "#fff59b",
      "borderRadius": 11.25,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 0,
      "position": "absolute",
      "top": undefined,
      "zIndex": 2,
    }
  }
>
  <Text
    maxFontSizeMultiplier={2.5}
    style={
      Object {
        "color": "#fff59b",
        "fontSize": 30.595932917809698,
        "paddingHorizontal": 15,
        "paddingVertical": 13.34,
        "textAlign": "center",
      }
    }
  >
    Sorry, you can't do that right now, you appear to be offline
  </Text>
</View>
`;
